lib/CGameState.cpp

```diff
@@ -1838,15 +1839,15 @@ void CGameState::initTowns()
 			int sel = -1;
 
 			for(ui32 ps=0;ps<vti->possibleSpells.size();ps++)
-				total += vti->possibleSpells[ps].toSpell()->getProbability(vti->subID);
+				total += vti->possibleSpells[ps].toSpell()->getProbability(FactionID(vti->subID));
```

Comment:
CGTownInstance might benefit from `getFaction` method that returns FactionID(subID), if only for clarity.

Reply:
@IvanSavenko so, it should inherit from INativeTerrainProvider?

Reply:
Don't know. Maybe?

Or simply use vti->town->faction instead of subID, should have the same effect.

My goal is to minimize usages of subID's for objects. And here vti->town->faction might be actually better than casting subID.

Reply:
done